src/lightly_train/_commands/common_helpers.py

```diff
@@ -391,18 +392,82 @@ def get_dataset_temp_mmap_path(
     """
     out_hash = get_sha256(f"{data}-{distributed_helpers.get_node_rank() or 0}")
     mmap_filepath = (cache.get_data_cache_dir() / out_hash).with_suffix(".mmap")
+    ref_count_filepath = mmap_filepath.with_suffix(".ref_count")
+
     mmap_filepath.parent.mkdir(parents=True, exist_ok=True)
 
-    reuse_file = Env.LIGHTLY_TRAIN_MMAP_REUSE_FILE.value
     try:
-        # Delete the file if it already exists from a previous run.
-        if not reuse_file and distributed_helpers.is_local_rank_zero():
-            _unlink_and_ignore(mmap_filepath)
+        # Increment reference count atomically
+        if distributed_helpers.is_local_rank_zero():
+            _increment_ref_count(ref_count_filepath)
 
         yield mmap_filepath
     finally:
-        if not reuse_file and distributed_helpers.is_local_rank_zero():
-            _unlink_and_ignore(mmap_filepath)
+        # Decrement reference count and cleanup if zero
+        if distributed_helpers.is_local_rank_zero():
+            _decrement_and_cleanup_if_zero(mmap_filepath, ref_count_filepath)
+
+
+def _increment_ref_count(ref_file: Path) -> None:
+    lock_file = ref_file.with_suffix(".lock")
```

Comment:
Should we pass the lock file to the function? Then it is clear that the same lock is used in the increment and decrement functions.